File: examples/python/full_flow.py
#!/usr/bin/env python3
"""Create → search → context → delete using the example MemcoreClient."""

from __future__ import annotations

import os
import sys

sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))

from memcore_client import MemcoreApiError, MemcoreClient


def require_env(name: str) -> str:
    value = os.environ.get(name, "").strip()
    if not value:
        print(f"error: missing required environment variable: {name}", file=sys.stderr)
        sys.exit(1)
    return value


def main() -> int:
    base_url = require_env("MEMCORE_BASE_URL")
    api_key = require_env("MEMCORE_API_KEY")
    org_id = require_env("MEMCORE_ORG_ID")
    user_id = os.environ.get("MEMCORE_USER_ID", "user_demo").strip() or "user_demo"

    client = MemcoreClient(base_url, api_key, org_id)

    try:
        created = client.create_memory(
            user_id, "User prefers concise technical summaries."
        )
        memories = created.get("memories") or []
        memory_id = memories[0]["id"] if memories else None
        print(f"create status={created.get('status')} memories={len(memories)}")

        search = client.search_memories(user_id, "technical summaries", limit=5)
        print(f"search results={len(search.get('results') or [])}")

        context = client.build_context(
            user_id, "How should replies be written?", max_tokens=1000
        )
        print(
            f"context status={context.get('status')} "
            f"chars={len(context.get('context') or '')}"
        )

        if memory_id:
            deleted = client.delete_memory(user_id, memory_id)
            print(f"delete status={deleted.get('status')}")
        else:
            print("warning: no memory id returned; skipping delete", file=sys.stderr)
    except MemcoreApiError as exc:
        print(str(exc), file=sys.stderr)
        return 1

    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: examples/node/full-flow.js
#!/usr/bin/env node
/**
 * Create → search → context → delete using the example MemcoreClient.
 * Never prints MEMCORE_API_KEY.
 */

import { MemcoreApiError, MemcoreClient } from "./memcore-client.js";

function requireEnv(name) {
  const value = (process.env[name] || "").trim();
  if (!value) {
    console.error(`error: missing required environment variable: ${name}`);
    process.exit(1);
  }
  return value;
}

async function main() {
  const baseUrl = requireEnv("MEMCORE_BASE_URL");
  const apiKey = requireEnv("MEMCORE_API_KEY");
  const orgId = requireEnv("MEMCORE_ORG_ID");
  const userId = (process.env.MEMCORE_USER_ID || "user_demo").trim() || "user_demo";

  const client = new MemcoreClient({ baseUrl, apiKey, orgId });

  try {
    const created = await client.createMemory({
      userId,
      text: "User prefers concise technical summaries.",
    });
    const memories = created.memories || [];
    const memoryId = memories[0]?.id;
    console.log(`create status=${created.status} memories=${memories.length}`);

    const search = await client.searchMemories({
      userId,
      query: "technical summaries",
      limit: 5,
    });
    console.log(`search results=${(search.results || []).length}`);

    const context = await client.buildContext({
      userId,
      query: "How should replies be written?",
      maxTokens: 1000,
    });
    console.log(
      `context status=${context.status} chars=${(context.context || "").length}`,
    );

    if (memoryId) {
      const deleted = await client.deleteMemory({ userId, memoryId });
      console.log(`delete status=${deleted.status}`);
    } else {
      console.error("warning: no memory id returned; skipping delete");
    }
  } catch (error) {
    if (error instanceof MemcoreApiError) {
      console.error(String(error));
      process.exit(1);
    }
    throw error;
  }
}

main();
